googletest/src/gtest-internal-inl.h

```diff
@@ -1074,24 +1085,44 @@ class StreamingListener : public EmptyTestEventListener {
   class SocketWriter : public AbstractSocketWriter {
    public:
     SocketWriter(const std::string& host, const std::string& port)
+#if GTEST_OS_WINDOWS
+        : sockfd_(INVALID_SOCKET), host_name_(host), port_num_(port) {
+#else
         : sockfd_(-1), host_name_(host), port_num_(port) {
+#endif
       MakeConnection();
     }
 
     ~SocketWriter() override {
+#if GTEST_OS_WINDOWS
+      if (sockfd_ != INVALID_SOCKET) CloseConnection();
+#else
       if (sockfd_ != -1) CloseConnection();
+#endif
     }
```

Comment:
Can we have single invocation of CloseConnection after #endif

Reply:
I'll leave it up to the maintainers but I tend to think that separating a braceless body from the condition by preprocessor directives is more worth avoiding.